code/game/mecha/equipment/tools/work_tools.dm

```diff
@@ -461,3 +461,50 @@
 	//NC.mergeConnectedNetworksOnTurf()
 	last_piece = NC
 	return 1
+
+//Dunno where else to put this so shrug
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade
+	name = "Ripley MK-II Conversion Kit"
+	desc = "A pressurized canopy attachment kit for an Autonomous Power Loader Unit \"Ripley\" MK-I mecha, to convert it to the slower, but space-worthy MK-II design. This kit cannot be removed, once applied."
+	icon_state = "ripleyupgrade"
+
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade/can_attach(obj/mecha/working/ripley/M as obj)
+	if(M.type != /obj/mecha/working/ripley)
+		to_chat(loc, "<span class='warning'>This conversion kit can only be applied to APLU MK-I models.</span>")
+		return 0
+	if(M.cargo.len)
+		to_chat(loc, "<span class='warning'>[M]'s cargo hold must be empty before this conversion kit can be applied.</span>")
+		return 0
+	if(!M.maint_access) //non-removable upgrade, so lets make sure the pilot or owner has their say.
+		to_chat(loc, "<span class='warning'>[M] must have maintenance protocols active in order to allow this conversion kit.</span>")
+		return 0
+	if(M.occupant) //We're actualy making a new mech and swapping things over, it might get weird if players are involved
+		to_chat(loc, "<span class='warning'>[M] must be unoccupied before this conversion kit can be applied.</span>")
+		return 0
+	return 1
+
+/obj/item/mecha_parts/mecha_equipment/ripleyupgrade/attach(obj/mecha/M as obj)
+	var/obj/mecha/working/ripley/mkii/N = new /obj/mecha/working/ripley/mkii(get_turf(M),1)
+	if(!N)
+		return
+	qdel(N.cell)
+	N.cell = null
+	if (M.cell)
+		N.cell = M.cell
+		M.cell.forceMove(N)
+		M.cell = null
+	N.step_energy_drain = M.step_energy_drain //For the scanning module
+	N.armor = N.armor.setRating(energy = M.armor["energy"]) //for the capacitor
+	for(var/obj/item/mecha_parts/mecha_equipment/E in M.equipment) //Move the equipment over...
+		N.equipment += E
+		E.forceMove(N)
+		E.chassis = N
+		M.equipment -= E
```

Comment:
I want to sorta force them over because it's supposed to seem like it's the same mech. As long as the new mech is updated correctly and the equipment's chassis pointer is updated, the equipment doesn't seem to care.

Reply:
All it needs to break horribly is for equipment to use callbacks, this is just unsafe. If something depends on attachment order preserve that info then use it here.

Reply:
Annoyingly enough, just running E.detach() and then E.attach(M) doesn't actually move anything over, so all the equipment gets deleted. If I'm going to need to get detach and attach put in here to get this merged, I'm gonna have to work at it a bit.

Reply:
Alright, got it to work. I still need to remove the equipment manually from M, since I assume that's normally handled with the detach button functionality, but otherwise detaching and re-attaching works.

Turns out what I was worried about isn't a big deal, since it's the equipment's try_attach() function that actually handles whether or not an equipment is allowed to attach.